x-pack/plugins/endpoint/server/routes/resolver/utils/normalize.ts

```diff
@@ -7,7 +7,7 @@
 import { ResolverEvent, LegacyEndpointEvent } from '../../../../common/types';
 
 function isLegacyData(data: ResolverEvent): data is LegacyEndpointEvent {
-  return data.agent.type === 'endgame';
+  return data.agent?.type === 'endgame';
```

Comment:
The endpoint isn't sending the `agent.type` field yet so guard against a crash here